src/services/gemini.py

```diff
@@ -1,73 +1,84 @@
-
 import google.generativeai as genai
 import asyncio, os
 import logging
 import time
+from google.generativeai.types import HarmCategory, HarmBlockThreshold
 from ratelimit import limits, sleep_and_retry
+from utils.config import GEMINI_API_KEY
 
 # Setup logging
 logging.basicConfig(level=logging.ERROR, format="%(asctime)s, %(levelname)s: %(message)s")
 
-class GeminiPro:
+class Gemini:
     def __init__(self):
         # Load API key
-        api_key = os.environ.get("GEMINI_API_KEY")
-        if not api_key:
+        if not GEMINI_API_KEY:
             logging.error("GEMINI_API_KEY is missing from envrionment variables.")
             raise ValueError("GEMINI_API_KEY is undefined.")
-        genai.configure(api_key=api_key)
+        genai.configure(api_key=GEMINI_API_KEY)
+
+        # Load Gemini with specific generation and safety settings
+        generation_config = {
+            "temperature": 0.4,
+            "top_p": 1,
+            "top_k":32,
+            "max_output_tokens": 2048
+        }
+        safety_settings = {
+            HarmCategory.HARM_CATEGORY_HATE_SPEECH: HarmBlockThreshold.BLOCK_LOW_AND_ABOVE,
+            HarmCategory.HARM_CATEGORY_HARASSMENT: HarmBlockThreshold.BLOCK_LOW_AND_ABOVE
+        }
 
```

Comment:
Model generation config specified for temperature, top_p, top_k, and max_output_tokens. Safety settings for hate speech and harassment specified as well.